platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/LocalStackDaoImpl.java

```diff
@@ -0,0 +1,232 @@
+        } finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    @Override
+    public void remove(String id) {
+        requireNonNull(id, "Stack id required");
+        lock.writeLock().lock();
+        try {
+            stacks.remove(id);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public void update(StackImpl update) throws NotFoundException {
+        requireNonNull(update.getId(), "Stack required");
+        lock.writeLock().lock();
+        try {
+            final StackImpl target = stacks.get(update.getId());
+            if (target == null) {
+                throw new NotFoundException(format("Stack with %s was not found", update.getId()));
+            }
+
+            if (update.getName() != null) {
+                target.setName(update.getName());
+            }
+            if (update.getDescription() != null) {
+                target.setDescription(update.getDescription());
+            }
+            if (update.getScope() != null) {
+                target.setScope(update.getScope());
+            }
+            if (update.getCreator() != null) {
+                target.setCreator(update.getCreator());
+            }
+            if (update.getTags() != null) {
+                target.setTags(update.getTags());
+            }
+            if (update.getWorkspaceConfig() != null) {
+                target.setWorkspaceConfig(update.getWorkspaceConfig());
+            }
+            if (update.getSource() != null) {
+                target.setSource(update.getSource());
+            }
+            if (!update.getComponents().isEmpty()) {
+                target.setComponents(update.getComponents());
+            }
+            target.setIcon(update.getIcon());
+            if (update.getPermissions() != null) {
+                target.setPermissions(update.getPermissions());
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    @Override
+    public List<StackImpl> getByCreator(String creator, int skipCount, int maxItems) {
+        requireNonNull(creator, "Stack creator required");
+        lock.readLock().lock();
+        try {
+            return FluentIterable.from(stacks.values())
+                                 .skip(skipCount)
+                                 .filter(stack -> {
+                                     return creator.equals(stack.getCreator());
+                                 })
+                                 .limit(maxItems)
+                                 .toList();
+        } finally {
+            lock.readLock().unlock();
+        }
+    }
+
+    @Override
+    public List<StackImpl> searchStacks(List<String> tags, int skipCount, int maxItems) {
+        lock.readLock().lock();
+        try {
+            return FluentIterable.from(stacks.values())
+                                 .skip(skipCount)
+                                 .filter(new Predicate<DecoratedStack>() {
```

Comment:
In above method you use lambda, but here anonymouse class, maybe it would be better to use lambda also here?